frontend/src/app/components/wp-card-view/services/wp-card-drag-and-drop.service.ts

```diff
@@ -0,0 +1,200 @@
+import {Inject, Injectable, Injector} from '@angular/core';
+import {WorkPackageResource} from "core-app/modules/hal/resources/work-package-resource";
+import {ReorderQueryService} from "core-app/modules/boards/drag-and-drop/reorder-query.service";
+import {States} from "core-components/states.service";
+import {WorkPackageChangeset} from "core-components/wp-edit-form/work-package-changeset";
+import {IWorkPackageCreateServiceToken} from "core-components/wp-new/wp-create.service.interface";
+import {WorkPackageCreateService} from "core-components/wp-new/wp-create.service";
+import {WorkPackageNotificationService} from "core-components/wp-edit/wp-notification.service";
+import {CurrentProjectService} from "core-components/projects/current-project.service";
+import {WorkPackageInlineCreateService} from "core-components/wp-inline-create/wp-inline-create.service";
+import {RequestSwitchmap} from "core-app/helpers/rxjs/request-switchmap";
+import {DragAndDropService} from "core-app/modules/boards/drag-and-drop/drag-and-drop.service";
+import {DragAndDropHelpers} from "core-app/modules/boards/drag-and-drop/drag-and-drop.helpers";
+import {WorkPackageCardViewComponent} from "core-components/wp-card-view/wp-card-view.component";
+
+@Injectable()
+export class WorkPackageCardDragAndDropService {
+
+  private _workPackages:WorkPackageResource[];
+
+  /** Whether the card view has an active inline created wp */
+  public activeInlineCreateWp?:WorkPackageResource;
+
+  /** A reference to the component in use, to have access to the current input variables */
+  public cardView:WorkPackageCardViewComponent;
+
+
+  // We remember when we want to update the query with a given order
+  public queryUpdates = new RequestSwitchmap(
```

Comment:
This will be fun when merging into `dev` due to all the changes in the drag and drop service :fearful:

Reply:
Yes, I already hate myself for that to be honest 😅 But it was really about time that we move this out of the component..